compiler/luci/partition/src/CircleOpCode.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CircleOpCode.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/IR/CircleNodeVisitor.h>
+
+#include <mio/circle/schema_generated.h>
+
+namespace
+{
+
+using namespace luci;
+using namespace circle;
+
+class QueryOpCode final : public CircleNodeVisitor<BuiltinOperator>
+{
+public:
+  // TODO add CircleNodes
+  BuiltinOperator visit(const CircleRsqrt *) final { return BuiltinOperator_RSQRT; }
+  BuiltinOperator visit(const CircleSqrt *) final { return BuiltinOperator_SQRT; }
```

Comment:
(optional) It would be good to make this not require operator-level implementation.

For example,
```cpp
const std::string toString(luci::CircleOpcode opcode)
{
  static const char *names[] = {
#define CIRCLE_NODE(OPCODE, CIRCLE_CLASS) #CIRCLE_CLASS,
#include <luci/IR/CircleNodes.lst>
#undef CIRCLE_NODE
  };
  // Returned string is the substring of circle class name ("luci::Circle" is sliced out)
  return std::string(names[static_cast<int>(opcode)]).substr(12);
}
```

Reply:
`circle::EnumNameBuiltinOperator()` already implements `toString()` :)

Reply:
`BuiltinOperator visit(const CircleXXX *)` is added individually for now and later can be refactored to shorter code...

Reply:
We already have `luci::CircleOpcode` which can be accessed as `circle_node.opcode()`
Is there any reason not to use it?
I think dependency with `BuiltinOperator_*` should be avoided if possible...!

Reply:
In draft, I just found that `luci::opcode_name` is just used for key.
So my point was "Can't we use `luci::CircleOpcode` as key?" :)

Reply:
Event I've make `luci::CircleOpcode`, forgot about this !
Thanks to remind me. I'll revise this PR :)

Reply:
I've found that KEY is from the USER as a TEXT STRING... in the `circle-partitioner` code... -_-;;;